### PR TITLE
fix(storage-resize-images): switched imagemagick for sharp library

### LINKED ISSUE #199: [storage-resize-images] WebP support
Seems `MIME type` `image/webp` is not support.

```
Error when resizing image { ChildProcessError: `convert /tmp/userDatas/6SaZVbNpTFOwwezEmtIc52ScLYw1/room_covers/room_cover_1581643929734.webp -resize 200x200> /tmp/room_cover_1581643929734_200x200.webp` failed with code 1
    at ChildProcess.<anonymous> (/srv/node_modules/child-process-promise/lib/index.js:132:23)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:915:16)
    at Socket.stream.socket.on (internal/child_process.js:336:11)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at Pipe._handle.close [as _onclose] (net.js:561:12)
  name: 'ChildProcessError',
  code: 1,
  childProcess: 
   ChildProcess {
     domain: 
      Domain {
        domain: null,
        _events: [Object],
        _eventsCount: 1,
        _maxListeners: undefined,
        members: [Array] },
     _events: { error: [Function], close: [Function] },
     _eventsCount: 2,
     _maxListeners: undefined,
     _closesNeeded: 3,
     _closesGot: 3,
     connected: false,
     signalCode: null,
     exitCode: 1,
     killed: false,
     spawnfile: 'convert',
     _handle: null,
     spawnargs: 
      [ 'convert',
        '/tmp/userDatas/6SaZVbNpTFOwwezEmtIc52ScLYw1/room_covers/room_cover_1581643929734.webp',
        '-resize',
        '200x200>',
        '/tmp/room_cover_1581643929734_200x200.webp' ],
     pid: 14,
     stdin: 
      Socket {
        connecting: false,
        _hadError: false,
        _handle: null,
        _parent: null,
        _host: null,
        _readableState: [Object],
        readable: false,
        domain: [Object],
        _events: [Object],
        _eventsCount: 2,
        _maxListeners: undefined,
        _writableState: [Object],
        writable: false,
        allowHalfOpen: false,
        _bytesDispatched: 0,
        _sockname: null,
        _pendingData: null,
        _pendingEncoding: '',
        server: null,
        _server: null,
        _idleNext: null,
        _idlePrev: null,
        _idleTimeout: -1,
        [Symbol(asyncId)]: 3152,
        [Symbol(bytesRead)]: 0 },
     stdout: 
      Socket {
        connecting: false,
        _hadError: false,
        _handle: null,
        _parent: null,
        _host: null,
        _readableState: [Object],
        readable: false,
        domain: [Object],
        _events: [Object],
        _eventsCount: 3,
        _maxListeners: undefined,
        _writableState: [Object],
        writable: false,
        allowHalfOpen: false,
        _bytesDispatched: 0,
        _sockname: null,
        _pendingData: null,
        _pendingEncoding: '',
        server: null,
        _server: null,
        _idleNext: null,
        _idlePrev: null,
        _idleTimeout: -1,
        write: [Function: writeAfterFIN],
        [Symbol(asyncId)]: 3153,
        [Symbol(bytesRead)]: 0 },
     stderr: 
      Socket {
        connecting: false,
        _hadError: false,
        _handle: null,
        _parent: null,
        _host: null,
        _readableState: [Object],
        readable: false,
        domain: [Object],
        _events: [Object],
        _eventsCount: 3,
        _maxListeners: undefined,
        _writableState: [Object],
        writable: false,
        allowHalfOpen: false,
        _bytesDispatched: 0,
        _sockname: null,
        _pendingData: null,
        _pendingEncoding: '',
        server: null,
        _server: null,
        _idleNext: null,
        _idlePrev: null,
        _idleTimeout: -1,
        write: [Function: writeAfterFIN],
        [Symbol(asyncId)]: 3154,
        [Symbol(bytesRead)]: 354 },
     stdio: [ [Object], [Object], [Object] ] },
  stdout: '',
  stderr: 'convert-im6.q16: delegate failed `\'dwebp\' -pam \'%i\' -o \'%o\'\' @ error/delegate.c/InvokeDelegate/1919.\nconvert-im6.q16: unable to open file `/tmp/magick-14rI_0cWl78wVr\': No such file or directory @ error/constitute.c/ReadImage/544.\nconvert-im6.q16: no images defined `/tmp/room_cover_1581643929734_200x200.webp\' @ error/convert.c/ConvertImageCommand/3258.\n' }
```

### PR DESCRIPTION
**Pull request by russellwheatley**
Fixes #199 

ImageMagick binary provided by Cloud Functions do not support `webp` image format.

### CODE REVIEW

**Comment by russellwheatley** [comment]
@laurenzlong  Happy to close this PR if `sharp` library isn't a desirable option. Here's a couple of links that helped my decision to use it:
[imagmagick doesn't support webp](https://stackoverflow.com/a/51658178)
[firebase function code sample](https://github.com/firebase/functions-samples/tree/master/image-sharp)

**Comment by laurenzlong** [comment]
Thanks for the context! We'll review it with the team, but it makes sense to me, especially since webp is what Google recommends as an image format for the web: https://developers.google.com/speed/webp

**Comment by karayu** [comment]
Thanks for putting this together! Let's get this out :)

**Comment by laurenzlong** [comment]
@russellwheatley Can you please look at PREINSTALL and POSTINSTALL and extension.yaml, and see if we are making explicit reference to ImageMagick, and if so, indicate that we are using Sharp not ImageMagick?

File: storage-resize-images/functions/src/index.ts [hunk 1]
```diff
@@ -39,10 +39,6 @@ admin.initializeApp();
 
 logs.init();
 
-/**
- * When an image is uploaded in the Storage bucket We generate a resized image automatically using
- * ImageMagick which is installed by default on all Cloud Functions instances.
- */
```
Comment (laurenzlong) [inline storage-resize-images/functions/src/index.ts:45]: instead removing the comment, can you update it to reflect that it is using Sharp?

**Pull request merged**